datasource.go

```diff
@@ -118,3 +128,75 @@ func (ds *sqldatasource) CheckHealth(ctx context.Context, req *backend.CheckHeal
 		Message: "Data source is working",
 	}, nil
 }
+
+func handleError(rw http.ResponseWriter, err error) {
+	rw.WriteHeader(http.StatusBadRequest)
+	_, err = rw.Write([]byte(err.Error()))
+	if err != nil {
+		backend.Logger.Error(err.Error())
+	}
+}
+
+type columnRequest struct {
+	Table string `json:"table"`
+}
+
+func (ds *sqldatasource) handle(resource string) func(rw http.ResponseWriter, req *http.Request) {
+	return func(rw http.ResponseWriter, req *http.Request) {
+		if ds.Completable == nil {
+			handleError(rw, errors.New("not implemented"))
+			return
+		}
+
+		var res []string
+		var err error
+		switch resource {
+		case "tables":
+			res, err = ds.Completable.Tables(req.Context(), ds.db)
+		case "schemas":
+			res, err = ds.Completable.Schemas(req.Context(), ds.db)
+		case "columns":
+			if req.Body == nil {
+				handleError(rw, errors.New("missing table name in request body"))
+				return
+			}
+			reqBody, err := ioutil.ReadAll(req.Body)
+			if err != nil {
+				handleError(rw, err)
+				return
+			}
+			column := columnRequest{}
+			err = json.Unmarshal(reqBody, &column)
+			if err != nil {
+				handleError(rw, err)
+				return
+			}
```

Comment:
This can be reduced,
```suggestion
			column := columnRequest{}
			if err := json.NewDecoder(req.Body).Decode(&column) {
				handleError(rw, err)
				return
			}
```